services/pipeline/src/main/java/tech/pegasys/pantheon/services/pipeline/PipelineBuilder.java

```diff
@@ -189,7 +189,7 @@ public PipelineBuilder(
             pipeEnd,
             maximumBatchSize,
             outputCounter.labels(lastStageName + "_outputPipe", "batches")),
-        bufferSize / maximumBatchSize + 1,
+        (int) Math.ceil(((double) bufferSize) / maximumBatchSize),
```

Comment:
Fixes the calculation so that a pipeline with buffer size 800, creating batches of 200 will hold 4 batches in the buffers (max 800 items in total).  The previous integer calculation would wind up buffering up to 5 batches.